cmd/utils/cmd.go

```diff
@@ -185,32 +190,87 @@ func ExportChain(blockchain *core.BlockChain, fn string) error {
 		defer writer.(*gzip.Writer).Close()
 	}
 
-	if err := blockchain.Export(writer); err != nil {
+	var z *zip.Writer
+	if strings.HasSuffix(fn, ".zip") {
+		z = zip.NewWriter(fh)
+		writer, err = z.Create("blocks.dat")
+		if err != nil {
+			return err
+		}
+	}
+
+	log.Info("Writing blocks", "first", first, "last", last)
+	if err := blockchain.ExportN(writer, first, last); err != nil {
 		return err
 	}
-	log.Info("Exported blockchain", "file", fn)
 
+	if z != nil {
+		log.Info("Writing state", "blockNumber", last)
+		writer, err := z.Create("state.dat")
+		if err != nil {
+			return err
+		}
+		if err := exportState(blockchain, chainDb, writer, last); err != nil {
+			return err
+		}
+
+		log.Info("Writing receipts", "blockNumber", last)
+		writer, err = z.Create("receipts.dat")
+		if err != nil {
+			return err
+		}
+		if err := exportReceipts(blockchain, chainDb, writer, last); err != nil {
+			return err
+		}
+
+		err = z.Close()
+		if err != nil {
+			return err
+		}
+	}
```

Comment:
It's not three lines if you refactor it into separate methods, though - all the boilerplate around opening the file and potentially compressing it would need duplicating too.